src/build/arg/mod.rs

```diff
@@ -45,6 +45,7 @@ type ValidatorOs<'a> = dyn FnMut(&OsStr) -> Result<(), Box<dyn Error + Send + Sy
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub(crate) enum ArgProvider {
     Generated,
+    GeneratedMutated,
```

Comment:
So the reason adding GeneratedMutated is prevent generated arg override generated mutated arg in the subcommand?

Then I wonder why we need to override (remove and readd) generated arg to generated (not mutated) arg in the subcomand.

Reply:
> override (remove and readd) generated arg to generated (not mutated) arg in the subcomand

We need to because all these generated args are global. We are not specially doing this for only generated args. We do this for all `global` args.

So, if we see global args, we pass them to subcommands. But if it's a special generated arg, they we sometimes decide not to pass it and skip it.